Updated reveal.js to v 4.10

koutbo6 opened the pull request:
### Purpose

Update reveal.js to v4.10 and include some additional useful plugints, addresses #1802 
Update also adds chalkboard and menu plugins.

### Documentation

Please note while this update works, it requires the following:
1- testing that mathjax and speaker notes work (currently works on my setup, mac + safari)
2- I think it is possible to make presentations configurable from front matter of slides. I am not familiar with hugo templating to make this work, and hope someone can give guidance or assist with this.

rodrigoalcarazdelaosa commented:
Very excited about this. I'll try to test it as much as I can

rodrigoalcarazdelaosa commented:
I made a quick test of this and it seems `shortcodes` don't work. I tried to add a figure in the [example `section.md`](https://github.com/koutbo6/wowchemy-hugo-modules/blob/reveal4/wowchemy/test/content/home/section.md) and fine, but if I add the same figure using the same code I see nothing in the slide. I also tried with a Youtube video and a custom `section.html` shortcake I had and nothing, that's why I think it has to do with shortcodes in general. Could you please take a look @koutbo6? Thanks a lot for this contribution!

koutbo6 commented:
@rodrigoalcarazdelaosa just to be clear ... are shortcodes broken in this pr? or shortcode dont work in slides only?

Can you confirm that shortcodes did work in slides before? can you give an example broken shortcode to make sure I view the same thing that you do

rodrigoalcarazdelaosa commented:
> @rodrigoalcarazdelaosa just to be clear ... are shortcodes broken in this pr? or shortcode dont work in slides only?
> 
> 
> 
> Can you confirm that shortcodes did work in slides before? can you give an example broken shortcode to make sure I view the same thing that you do

I added a figure in the example `section.md` of the test site and it did work. I added the same shortcode in a slide (using the reveal4 branch) and it didn't work (I saw nothing and inspecting the HTML I couldn't see the figure). This is the [live version](https://rodrigoalcarazdelaosa.me/slides/fuerzas-2eso/) of the slide I was testing (this one is using reveal 3 and you see images and videos working fine).

koutbo6 commented:
@rodrigoalcarazdelaosa I can confirm youtube and icon shortcodes work in slides on my end ... would appreciate a concrete example that fails on your end.

here is the index.md that I am using which works perfectly .... how can we debug this and make sure we have the same configuration? .....

koutbo6 commented:
let me create a new clone and site from the pr and try to replicate what you did ... if u can share some slide source that u know doesnt work .. that would help

rodrigoalcarazdelaosa commented:
> @rodrigoalcarazdelaosa I can confirm youtube and icon shortcodes work in slides on my end ... would appreciate a concrete example that fails on your end.
> 
> 
> 
> here is the index.md that I am using which works perfectly .... how can we debug this and make sure we have the same configuration? .....
> 
> 

Are you using the `test` site shipped with Wowchemy?

koutbo6 commented:
@rodrigoalcarazdelaosa I did the following and it worked for me:

1- Cloned reveal4 branch from koutbo64
2- Cd to test directory, then created slides/example/ directories
3- Created my index.md for slides under example directory 
4- started the hugo server and opened path slides/example ... and it worked

I noticed that if I dont follow this directory structure for slides where I place them under the root "slides" directory .. then slides will not render correctly

But everything else seems to work fine .. for math and diagrams you have to enable math: true and diagrams: true in front matter or set them to true in params.toml

I'm using latest hugo version on a mac .... any feedback welcome

koutbo6 commented:
just pushed the test slides I used under test:

/test will show slides that dont render correctly
/slides/example works fine

Chrede88 commented:
I've also been looking into `shortcodes`. So far only been looking at `fragments`. That one is absolutely broken!
The markdown parser must have changed, because it seems that raw html is simply **ignored** now. I'm looking at the slides used in the `academic-starter` repo, `./exampleSite/content/slides/example/index.md`. Looking at slide 6, the rendered html source is:
```html
<section>
<h2 id="fragments">Fragments</h2>
<p>Make content appear incrementally</p>
<pre><code>{{% fragment %}} One {{% /fragment %}}
{{% fragment %}} **Two** {{% /fragment %}}
{{% fragment %}} Three {{% /fragment %}}
</code></pre><p>Press <code>Space</code> to play!</p>
<p><!-- raw HTML omitted -->
One
<!-- raw HTML omitted -->
<!-- raw HTML omitted -->
<strong>Two</strong>
<!-- raw HTML omitted -->
<!-- raw HTML omitted -->
Three
<!-- raw HTML omitted --></p>
</section>
```

Looking at the output of the `fragment` shortcode:
```html
<span class="fragment {{ .Get "class" }}" {{ with .Get "weight" }}data-fragment-index="{{ . }}"{{ end }}>
  {{ .Inner }}
</span>
```
Which is far from what reveal is now expecting (https://revealjs.com/markdown/#element-attributes):
```html
<section data-markdown>
  <script type="text/template">
    - Item 1 <!-- .element: class="fragment" data-fragment-index="2" -->
    - Item 2 <!-- .element: class="fragment" data-fragment-index="1" -->
  </script>
</section>
```

This suggests that all shortcodes that simply returns `html` is no longer usable! And so, we would need to rewrite all current "slides only" shortcodes and make special slides shortcodes for things like figures, which are not a slides specific shortcode.

Chrede88 commented:
Other shortcodes I've checked out:

1) `speaker_note` shortcode still works.
2) `slide` shortcode is broken.

Chrede88 commented:
Long code blocks can't be scrolled without also moving through slides.
We need a to add `data-prevent-swipe` to code blocks.
```html
<section>
  <p data-prevent-swipe>
    Can't change slides by swiping across this element.
  </p>
</section>
```

rodrigoalcarazdelaosa commented:
Relevant: https://revealjs.com/markdown/

rodrigoalcarazdelaosa commented:
> This suggests that all shortcodes that simply returns html is no longer usable! And so, we would need to rewrite all current "slides only" shortcodes and make special slides shortcodes for things like figures, which are not a slides specific shortcode.

If only there was a way to detect if a shortcode is being called from inside a slide, we could then add slide-specific code

rodrigoalcarazdelaosa commented:
> > This suggests that all shortcodes that simply returns html is no longer usable! And so, we would need to rewrite all current "slides only" shortcodes and make special slides shortcodes for things like figures, which are not a slides specific shortcode.
> 
> If only there was a way to detect if a shortcode is being called from inside a slide, we could then add slide-specific code

Probably using the `.Parent` variable

koutbo6 commented:
> > > This suggests that all shortcodes that simply returns html is no longer usable! And so, we would need to rewrite all current "slides only" shortcodes and make special slides shortcodes for things like figures, which are not a slides specific shortcode.
> > 
> > 
> > If only there was a way to detect if a shortcode is being called from inside a slide, we could then add slide-specific code
> 
> Probably using the `.Parent` variable

Why not use jquery in baseof.html to inject these directives to relevant tags . ... 

Think we can automate testing for this issue? at the very least, let's have a unifide testcase that shows all these issues so we can manually check, and maybe a checklist of what is broken?

koutbo6 commented:
> Other shortcodes I've checked out:
> 
> 1. `speaker_note` shortcode still works.
> 2. `slide` shortcode is broken.

can confirm that as well ..

Btw .. i created a copy of the starter slides under /test/
http://localhost:1313/slides/example/

I'm going to modify slightly to illustrate all the issues we know so far

If you come across another issue not there ... please let me know or assist by updating the slides to reflect it
Consider the slides our checklist for now for what needs to be fixed

Thank you all for the feedback

koutbo6 commented:
> Long code blocks can't be scrolled without also moving through slides.
> We need a to add `data-prevent-swipe` to code blocks.
> 
> ```
> <section>
>   <p data-prevent-swipe>
>     Can't change slides by swiping across this element.
>   </p>
> </section>
> ```

maybe this is related to enabling mousewheel scrolling? can u try disabling it (see front matter configuration in test slide) and see if that helps

would appreciate a copy of what you are testing to make sure we are on the same page ... is the code lengthy on vertical or horizontal axes?

koutbo6 commented:
@rodrigoalcarazdelaosa @Chrede88 

managed to solve the omitted HTML problem by adding the following to config.yaml for the test site:

```yaml
markup:
  goldmark:
    renderer:
      unsafe: true 
```

more information available here:

https://stackoverflow.com/questions/63198652/hugo-shortcode-ignored-saying-raw-html-omitted

I also noticed that this resulted in callout shortcode working fine for me for regular pages, which didnt work before

Note sure whether there is a security concern or not there.

koutbo6 commented:
fragements worked in slides after this change

rodrigoalcarazdelaosa commented:
> @rodrigoalcarazdelaosa @Chrede88 
> 
> 
> 
> managed to solve the omitted HTML problem by adding the following to config.yaml for the test site:
> 
> 
> 
> ```yaml
> 
> markup:
> 
>   goldmark:
> 
>     renderer:
> 
>       unsafe: true 
> 
> ```
> 
> 
> 
> more information available here:
> 
> 
> 
> https://stackoverflow.com/questions/63198652/hugo-shortcode-ignored-saying-raw-html-omitted
> 
> 
> 
> I also noticed that this resulted in callout shortcode working fine for me for regular pages, which didnt work before
> 
> 
> 
> Note sure whether there is a security concern or not there.

That's the [default configuration](https://github.com/wowchemy/starter-academic/blob/6eb6521dfa40f0fbae42d698da0f40846a0b4cba/config/_default/config.toml#L57) for normal templates, not sure why @gcushen didn't include it in the test site

koutbo6 commented:
code code highlighting .. html seems to be generated correctly to highlight code
However, it is then escaped ... afraid I'll have to stay away from this for a few days
if anyone can suggest the cause of why html is being escaped and can suggest at least where to look to fix this... would help greatly

rodrigoalcarazdelaosa commented:
> @rodrigoalcarazdelaosa @Chrede88
> 
> managed to solve the omitted HTML problem by adding the following to config.yaml for the test site:
> 
> ```yaml
> markup:
>   goldmark:
>     renderer:
>       unsafe: true 
> ```
> 
> more information available here:
> 
> https://stackoverflow.com/questions/63198652/hugo-shortcode-ignored-saying-raw-html-omitted
> 
> I also noticed that this resulted in callout shortcode working fine for me for regular pages, which didnt work before
> 
> Note sure whether there is a security concern or not there.

Further testing with the latest version. Now I can see Youtube videos and figures but for example I'm no seeing `<iframe>`, try this one (which works with the current version).
```
<iframe src="https://phet.colorado.edu/sims/html/masses-and-springs-basics/latest/masses-and-springs-basics_es.html" width="100%" height="500" scrolling="no" allowfullscreen></iframe>
```
A [custom shortcode for sections](https://github.com/rodrigoalcarazdelaosa/rodrigoalcarazdelaosa.me/blob/master/layouts/shortcodes/section.html) that I had is not working either.

koutbo6 commented:
ok i think the issue is not with the new code but the configuration of the test server

I managed to get @rodrigoalcarazdelaosa iframe code to work on my academic starter website perfectly .. and all the issues with the code highlighting are resolved there without making any changes

to confirm this try the following:

run the iframe example on test site shipped with wowchemy on master branch and on reveal4 branch
You will notice both will not work .. this tells us it is not to do with our code but something else in the configuration

To further confirm this, create a new academic starter template site and point the go.mod to the reveal4 branch of wochemy and test slides there ... everything works perfectly

Its clear to me now that the issue is likely hugo related rather than reveal .. we just need to flip the config switches that will fix our problems

I'll keep u posted

rodrigoalcarazdelaosa commented:
> ok i think the issue is not with the new code but the configuration of the test server
> 
> I managed to get @rodrigoalcarazdelaosa iframe code to work on my academic starter website perfectly .. and all the issues with the code highlighting are resolved there without making any changes
> 
> to confirm this try the following:
> 
> run the iframe example on test site shipped with wowchemy on master branch and on reveal4 branch
> You will notice both will not work .. this tells us it is not to do with our code but something else in the configuration
> 
> To further confirm this, create a new academic starter template site and point the go.mod to the reveal4 branch of wochemy and test slides there ... everything works perfectly
> 
> Its clear to me now that the issue is likely hugo related rather than reveal .. we just need to flip the config switches that will fix our problems
> 
> I'll keep u posted

Interesting. Perhaps @gcushen wants to revise the `test` site configuration so the tests we perform there are meaningful and compare well to real life situations. I'll make sure to test your branch with a `starter-academic` template

koutbo6 commented:
Give me a few minutes ... I think I'll have a better working test side this time

rodrigoalcarazdelaosa commented:
> Give me a few minutes ... I think I'll have a better working test side this time

Great. I'll also test my already created slides to check nothing weird appears

Chrede88 commented:
> > Long code blocks can't be scrolled without also moving through slides.
> > We need a to add `data-prevent-swipe` to code blocks.
> > ```
> > <section>
> >   <p data-prevent-swipe>
> >     Can't change slides by swiping across this element.
> >   </p>
> > </section>
> > ```
> 
> maybe this is related to enabling mousewheel scrolling? can u try disabling it (see front matter configuration in test slide) and see if that helps
> 
> would appreciate a copy of what you are testing to make sure we are on the same page ... is the code lengthy on vertical or horizontal axes?

It has to do with `touch` navigation: https://revealjs.com/touch-navigation/
The problem comes up when code blocks extend beyond the page bounds horizontally.

koutbo6 commented:
@Chrede88 check the front matter .. i think u can disable it there to fix this behavior
just a flag .. I'll check it in my next update to the test site

Chrede88 commented:
> @Chrede88 check the front matter .. i think u can disable it there to fix this behavior
> just a flag .. I'll check it in my next update to the test site

Right! I think touch navigation is important:) I don't want to get rid of it, but just make sure code blocks are scrollable ;)

koutbo6 commented:
ok this part was the culprit from config file (im using toml here not yaml):

```toml
  [markup.highlight]
    codeFences = false 
```

koutbo6 commented:
ok please add this to config.yaml to fix all previously mentioned issues:

```yaml
markup:
  defaultMarkdownHandler: goldmark
  goldmark:
    renderer:
      unsafe: true
  highlight:
    codeFences: false
```

koutbo6 commented:
should fix:
- iframe
- code highlighting
- slide shortcode

I'll update the code to also add touch and loop frontmatter options and correct the config file

koutbo6 commented:
please check the updated test site

improvements include adding loop and touch configurations to slide front matter

also please visit /test when u run the test site

you can see all the shortcodes that work in regular pages, but not slides even in older versions of wowchemy

Do we want to fix those?

I would like to see diagrams and charts work ... any thoughts on how to achieve that?

maybe another pr after this one is tested and merged

Thanks for the supprt!

koutbo6 commented:
> > @Chrede88 check the front matter .. i think u can disable it there to fix this behavior
> > just a flag .. I'll check it in my next update to the test site
> 
> Right! I think touch navigation is important:) I don't want to get rid of it, but just make sure code blocks are scrollable ;)

@Chrede88 , would surrounding long code with the following resolve the issue as the link you posted suggests?

```
<p data-prevent-swipe>

// code here

</p>
```

Chrede88 commented:
> > > @Chrede88 check the front matter .. i think u can disable it there to fix this behavior
> > > just a flag .. I'll check it in my next update to the test site
> > 
> > 
> > Right! I think touch navigation is important:) I don't want to get rid of it, but just make sure code blocks are scrollable ;)
> 
> @Chrede88 , would surrounding long code with the following resolve the issue as the link you posted suggests?
> 
> ```
> <p data-prevent-swipe>
> 
> // code here
> 
> </p>
> ```

Yes:)

dralmarzouq commented:
add the following slides from matter configurations:

diagram: to enable or disable mermaid diagrams
diagram_theme: to set the preferred diagram theme

enjoy

dralmarzouq commented:
mermaid diagram works in slides now

Problem was lazy loading of slide contents, there is no event fired when slide is loaded, so added some code to sense if we are close to a slide that is already loaded and will render mermaid 

Noticed the following:

1- printing pdfs will print the mermaid code rather than diagram, unless you view the slides containing the diagrams to render them
2- mermaid code is now sensitive to whitespace, empty lines will result in syntax error
3- Flowcharts sometime do not render nodes correctly, and this seems to be an issue with mermaid.

rodrigoalcarazdelaosa commented:
> mermaid diagram works in slides now
> 
> Problem was lazy loading of slide contents, there is no event fired when slide is loaded, so added some code to sense if we are close to a slide that is already loaded and will render mermaid
> 
> Noticed the following:
> 
> 1- printing pdfs will print the mermaid code rather than diagram, unless you view the slides containing the diagrams to render them
> 2- mermaid code is now sensitive to whitespace, empty lines will result in syntax error
> 3- Flowcharts sometime do not render nodes correctly, and this seems to be an issue with mermaid.

I see some issues with some of the diagrams but that's probably caused by the diagram itself? Or probably due to `mermaid` not playing nice with `reveal` (`upstream` then).

<img width="1440" alt="Captura de pantalla 2020-12-12 a las 19 43 39" src="https://user-images.githubusercontent.com/9049533/101992337-c0b71000-3cb2-11eb-94d7-abe6b240a7ea.png">

<img width="1440" alt="Captura de pantalla 2020-12-12 a las 19 43 58" src="https://user-images.githubusercontent.com/9049533/101992340-c6145a80-3cb2-11eb-8243-6f21229a105e.png">

rodrigoalcarazdelaosa commented:
Testing one of my already created slides I see that now the slide, when we reach the end, goes back to the beginning again. This also happens with [sections](https://github.com/rodrigoalcarazdelaosa/rodrigoalcarazdelaosa.me/blob/master/layouts/shortcodes/section.html) (moving vertically), which is a bit weird but probably default behavior with the new version?

koutbo6 commented:
@rodrigoalcarazdelaosa 
Most likely its a configuration issue for everything you mentioned .. and the rendering is mermaid related

You might want to try the new directives feature in mermaid:

```mermaid
{{% diagram %}}
%%{init: { 'logLevel': 'debug', 'theme': 'neutral' } }%%
graph TD;
%%{config: { 'fontFamily': 'Menlo', 'fontSize': 18, 'fontWeight': 400} }%%
    A-->B;
    A-->C;
    B-->D;
    C-->D;
{{% /diagram %}}
```

Try changing the configuration of the diagram see if you can get better results.
The issue so far is with the flowdiagram only

For the loop back, can you share the code to the presentation? I need to experience the error

rodrigoalcarazdelaosa commented:
> For the loop back, can you share the code to the presentation? I need to experience the error

You can try any of the slides I have [here](https://github.com/rodrigoalcarazdelaosa/rodrigoalcarazdelaosa.me/tree/master/content/es/slides). You'll need images and a custom `section.html` shortcode.

gcushen commented on wowchemy/assets/css/reveal_custom.css:
```diff
@@ -0,0 +1,16 @@
+// Mermaid.js diagram div
```
This file is intended for the site admin and should be left empty. Why have you duplicated CSS code here from another file?

dralmarzouq replied:
That leaked from my own site, should be removed in next push

gcushen commented on wowchemy/assets/js/wowchemy-theming.js:
```diff
@@ -173,10 +173,10 @@ function renderThemeVariation(isDarkTheme, init = false) {
       } else {
         // Have to reload to re-initialise Mermaid with the new theme and re-parse the Mermaid code blocks.
-        location.reload();
+        // location.reload();
```
Why have you commented out this line? Line 159 also performs a similar action. The page is reloaded for a reason, so are you suggesting the new version of Mermaid works without reload in this case - i.e. that the Mermaid theme can be dynamically updated without re-initialising?

dralmarzouq replied:
was figuring things out and I uncommented one but forgot the other :)

gcushen commented on wowchemy/layouts/shortcodes/diagram.html:
```diff
@@ -1,3 +1,4 @@
+
```
Please remove the line break you added

gcushen commented on wowchemy/layouts/slides/baseof.html:
```diff
@@ -20,55 +21,176 @@
+  {{- $loop := $.Param "slides.loop" | default true -}}
 
+<!-- The following is just to set the proper configurations based on slides front matter to initialize reveal -->
   <script>
```
Please refactor the JS to a dedicated file and use Hugo's new JS Param system rather than using Go templating without JS code.

gcushen commented on wowchemy/layouts/slides/baseof.html:
```diff
@@ -20,55 +21,176 @@
+
+<!-- 3rd part plugins -->
+
+  <script src="/js/vendor/reveal.js/plugin/menu/menu.js"></script>
```
Why are you suggesting to load some Reveal-related JS assets from CDN and some locally?

rodrigoalcarazdelaosa replied:
> Why are you suggesting to load some Reveal-related JS assets from CDN and some locally?

Both the `menu` and the `chalkboard` plugins are not built-in plugins are therefore are not available from a CDN. Ideally they could be optional (to load) but I really think they add some very interesting value.

dralmarzouq replied:
exactly what @rodrigoalcarazdelaosa said.

There is even a request to simplify adding reveal plugins ... might look into it as it seems very doable

will be a busy week .. expect delays

rodrigoalcarazdelaosa replied:
BTW, I'd prefer for the `menu` plugin to only show the menu by default, not also the possibility to select a different theme.

gcushen replied:
> > Why are you suggesting to load some Reveal-related JS assets from CDN and some locally?
> 
> Both the `menu` and the `chalkboard` plugins are not built-in plugins and therefore are not available from a CDN.

I've just done a quick search and the plugins you suggested do seem to be published in one of the popular JS CDNs: https://www.jsdelivr.com/package/npm/reveal.js-plugins

It would facilitate reviewing (and hopefully significantly reduce the number of new/changed files from **70**) if we could at least refactor this PR to only use CDN for assets.

gcushen replied:
As for plugins, I suggest to consider including some plugins which include common presentation functionality such as Speaker Notes, Menu, and Chalkboard by default.

In the future, we can consider adding something like `slide_plugins` in front matter to include additional plugins.

As for configuring slides and plugins, let's consider renaming `slides` to `slide_options` so that the namespace doesn't clash with the use of `slides` elsewhere for linking slides to pages. Then we can consider having sub-options under that for each plugin, similar to the front matter structure used in Rmarkdown: https://bookdown.org/yihui/rmarkdown/reveal-js-plugins.html

dralmarzouq replied:
renaming all options to use lowercase and underscore names instead of camel case. I hope we can keep this consistent for the future.

I like the blowdown approach for plugins and should be the way forward.

speaker notes is there by default. Its now shipped with reveal and enabled by default along with the following:

  <script src="{{ $cdn_url_reveal }}/reveal.min.js"></script>
  <script src="{{ $cdn_url_reveal }}/plugin/markdown/markdown.js"></script>
  <script src="{{ $cdn_url_reveal }}/plugin/highlight/highlight.js"></script>
  <script src="{{ $cdn_url_reveal }}/plugin/notes/notes.js"></script>
  <script src="{{ $cdn_url_reveal }}/plugin/search/search.js"></script>
  <script src="{{ $cdn_url_reveal }}/plugin/math/math.js"></script>
  <script src="{{ $cdn_url_reveal }}/plugin/zoom/zoom.js"></script>
chalkboard is a small plugin and the only version available on can is an old one that is not compatible with reveal4. So I kets the files in the static dir.

I did exclude menu plug in file since I was able to successfully add it from cdn. It seems to work perfectly, but the following error can be seen in the console:

SyntaxError: import.meta is only valid inside modules.

Not sure how that might impact slides, so please test and let me know.

in the slide front matter, the slides: namespace was already there to configure slide themes and transitions. It isn't something that I added. I just switched from hard coded configuration into a more dynamic one where the dictionary is read with all the configuration and the options passed to reveal and all other plugin.

Only challenge is defaults. I'll have to create an object in Wowchemy-slides.js which include some of the default values when the options are not set.

I think keeping it as slides: is cleaner, unless you can think of an instance where there indeed is a conflict. It will ensure that older slides will remain compatible with the new update.

I will add defaults to enable menu and chalkboard, but will keep the option to disable them in case someone runs into a compatibility issue with them on their slides.

pushing an update soon

dralmarzouq replied:
im also moving the menu and chalkboard options under reveal_options since they are plugins

gcushen replied:
Thanks, great progress on this!

As for local assets, please continue removing the remaining Reveal-related local assets (Chalkboard) from this PR and consider requesting the author to publish the Reveal v4 compatible version to the CDN if necessary.

gcushen replied:
@dralmarzouq in case you missed this comment ^

dralmarzouq replied:
removed chalkboard until it is hosted on cdn .. I can host on my own website if needed
only menu is now available by default.
online classes will not be the same without chalkboard

rodrigoalcarazdelaosa replied:
> removed chalkboard until it is hosted on cdn .. I can host on my own website if needed
> only menu is now available by default.
> online classes will not be the same without chalkboard

Hoping that [rajgoel](https://github.com/rajgoel) will update it in the CDN soon 💪🏼

dralmarzouq replied:
@rodrigoalcarazdelaosa would definitely help if you know him and can encourage him to do so ... I somehow doubt he will look at the ticket I submitted. The repo has been inactive for a while now.

rodrigoalcarazdelaosa replied:
> @rodrigoalcarazdelaosa would definitely help if you know him and can encourage him to do so ... I somehow doubt he will look at the ticket I submitted. The repo has been inactive for a while now.

I'm afraid I don't have any relationship with him but I have given a thumbs up in [your issue](https://github.com/rajgoel/reveal.js-plugins/issues/106). I'd change the title to _consider updating_ instead of _consider hosting_, as the package is hosted but not updated. If hosting the plugin locally works, it's just a matter of updating the plugin in the CDN, which shouldn't take much effort right?

dralmarzouq replied:
I did try using jsdeliver github mirroring cdn ... did not work for some reason.
You might have better success than me.
Worked only when hosted locally

dralmarzouq reviewed (commented):
renaming all options to use lowercase and underscore names instead of camel case. I hope we can keep this consistent for the future.

I like the blowdown approach for plugins and should be the way forward.

speaker notes is there by default. Its now shipped with reveal and enabled by default along with the following:

```html
  <script src="{{ $cdn_url_reveal }}/reveal.min.js"></script>
  <script src="{{ $cdn_url_reveal }}/plugin/markdown/markdown.js"></script>
  <script src="{{ $cdn_url_reveal }}/plugin/highlight/highlight.js"></script>
  <script src="{{ $cdn_url_reveal }}/plugin/notes/notes.js"></script>
  <script src="{{ $cdn_url_reveal }}/plugin/search/search.js"></script>
  <script src="{{ $cdn_url_reveal }}/plugin/math/math.js"></script>
  <script src="{{ $cdn_url_reveal }}/plugin/zoom/zoom.js"></script>
```

chalkboard is a small plugin and the only version available on can is an old one that is not compatible with reveal4. So I kets the files in the static dir.

I did exclude menu plug in file since I was able to successfully add it from cdn. It seems to work perfectly, but the following error can be seen in the console:

`SyntaxError: import.meta is only valid inside modules.`

Not sure how that might impact slides, so please test and let me know.


in the slide front matter, the slides: namespace was already there to configure slide themes and transitions. It isn't something that I added. I just switched from hard coded configuration into a more dynamic one where the dictionary is read with all the configuration and the options passed to reveal and all other plugin.

Only challenge is defaults. I'll have to create an object in Wowchemy-slides.js which include some of the default values when the options are not set.

I think keeping it as slides: is cleaner, unless you can think of an instance where there indeed is a conflict. It will ensure that older slides will remain compatible with the new update.

I will add defaults to enable menu and chalkboard, but will keep the option to disable them in case someone runs into a compatibility issue with them on their slides.

pushing an update soon

dralmarzouq commented:
@gcushen seems like Hugo modifies the case used in front matter configuration keys.

For example, slideNumber will be lower cased to slidenumber when accessed from the template parameter context variable.

Is there a way to maintain case? this is breaking configuration for any js plugins that uses camel case to name configuration keys.

Only solution I can think of is to use snake case and convert them into camel case.

dralmarzouq commented:
lots of changes in this one:

- menu now included from CDN, but not chalkboard
- configurations in front matter changed:
    + added mermaid_enabled under slides to enabled/disable mermaid, with default being disabled
    + added menu_enabled and chalkboard_enabled under reveal_options to enable/disable each feature, with default being enabled
    + configurations under reveal_options are converted into a js object, with keys that are snake case converted into camel case since this is what reveal.js uses. All plugins can also be configured by including their key and exactly like their documentation states. see the menu example.
- we now have a rudimentary way of adding reveal plugins. chalkboard can be used as an example to demonstrate how it works. By including js distribution under assets. Another options is to link to a CDN. Configuration could be passed from front matter. The only problem is how to initialize reveal. I'm not sure if the initialize method could be called multiple times or just a single time. But its something I have no urgency to fix now and could tackle in the future when I have time.
- Note that issues related to rendering diagrams seem to be relation to transitions. Slide and fade transitions work best with mermaid.
- Caught a few bugs, like mermaid would not render if you dont use transitions. Now it works with no transitions also. Cant remember the rest.

gcushen commented on wowchemy/archetypes/slides/index.md:
```diff
@@ -12,7 +12,42 @@ slides:
+  #   Use lower case names and replace space with hyphen '-'
+  highlight_style: monokai
+
+  mermaid_enabled: true # configure js or not
```
There is an existing front matter option for this in Wowchemy, named `diagram`: https://wowchemy.com/docs/writing-markdown-latex/#diagrams . Let's try to be consistent.

dralmarzouq replied:
mermaid configuration can also be passed exactly the same way as reveal configuration.
Didn't make send to have
diagram: true to enable diagrams
mermaid: to configure options

its now set such that:
mermaid_enabled: true/false
mermaid:  to pass configuration
similar to menu_enabled and menu, chalkboard_enabled and chalkboard

pick your preferred type of consistency

koutbo6 replied:
im going to change mermaid_option to diagram_options and explain in the comments that its mermaid options.
I'm not really sure if the diagram configuration is within scope of slides or not

gcushen commented on wowchemy/archetypes/slides/index.md:
```diff
@@ -12,7 +12,42 @@ slides:
-  highlight_style: dracula
+  #   Available highlight themes listed in: https://highlightjs.org/static/demo/
+  #   Use lower case names and replace space with hyphen '-'
+  highlight_style: monokai
```
What's the reason for changing the default highlight theme?

dralmarzouq replied:
the comments are there to direct user to the list of available highlights.

I copied what I had set while testing things in the test slide.

I can set back to default

gcushen commented on wowchemy/archetypes/slides/index.md:
```diff
@@ -12,7 +12,42 @@ slides:
+  mermaid_enabled: true # configure js or not
+  mermaid_options: # see mermaid documentation for other options
+    theme: base # default/base/dark/neutral/forest
+    securityLevel: loose
```
Let's use the same security level as already used in Wowchemy JS and not expose this to the end user.

dralmarzouq replied:
Wowchemy uses defaults .. it was changed in 8.2 to prevent clickable node unless you set security level to loose. Which was the default prior to 8.2.

Let's make it configurable but keep the default as strict

gcushen replied:
I don't follow why you suggest to expose the Mermaid security level to the user editing the Markdown slides - please can you clarify?

What is the issue with just setting a fixed Mermaid security level in the code?

gcushen commented on wowchemy/assets/js/wowchemy-slides.js:
```diff
@@ -0,0 +1,188 @@
+  });
+  // end of mermaid diagram functions
+
+  function fixMermaid() {
```
Did you duplicate this `fixMermaid` code from the main Wowchemy JS file?

dralmarzouq replied:
I like live dangerously .. any other suggestions? its just a single function I think.

It fixed the issue where the \```mermaid directive doesnt work

gcushen replied:
`fixMermaid()` has been refactored to a dedicated JS module `wowchemy-utils.js` now so that that specific function can be imported into this PR. Also, the function now uses vanilla JS rather than JQuery.

dralmarzouq replied:
@gcushen ok I have done what I can

Last change should address the last of your comments

if merging please take care of this part to use the Wowchemy-utils.js

I dont see it merged to master so I cannot use it. I need to also shift my attention somewhere else

thanks for the feedback

gcushen replied:
> if merging please take care of this part to use the Wowchemy-utils.js
> I dont see it merged to master so I cannot use it. I need to also shift my attention somewhere else

`wowchemy-utils.js` is located here: https://github.com/wowchemy/wowchemy-hugo-modules/blob/master/wowchemy/assets/js/wowchemy-utils.js

gcushen commented on wowchemy/layouts/slides/baseof.html:
```diff
@@ -20,55 +21,52 @@
+  <link rel="stylesheet" href="{{ $cdn_url_reveal }}/reveal.min.css">
   {{- $theme := $.Param "slides.theme" | default "black" -}}
-  <link rel="stylesheet" href="{{ $cdn_url_reveal }}/css/theme/{{ $theme }}.min.css">
+  <link rel="stylesheet" href="{{ $cdn_url_reveal }}/theme/{{ $theme }}.min.css" id="theme" />
```
If this ID is needed, consider naming it `reveal-theme` so it's clear which theme it's referring to elsewhere.

dralmarzouq replied:
needed by reveal menu plugin to identify the available list of themes. Cant change it im afraid

gcushen replied:
Consider requesting the author to add support for custom identifiers (or helping contribute to the library), otherwise this is open to conflicts (especially if it ends up embedded in other pages (https://revealjs.com/presentation-size/#embedded ), and may even conflict with multiple embedded slides on a page?) as well as causing confusion for future maintainers/contributors given the other themes in use.

koutbo6 replied:
removed it .. website authors need to mess with configuration in the frontend to get theme selection to work 
I think its still configurable so we are not losing much

koutbo6 commented:
@gcushen incorporated your last suggestions and cleaned up what I can find
mainly:
1- id=theme was removed along with code that reacted to it in wowchemy-slides.js
2- used the diagram=true variable and added diagram_options to pass on mermaid configuration.

I see you are refactoring fixMermaid function so I didnt mess with that. I'll leave that to you.

gcushen commented on wowchemy/layouts/slides/baseof.html:
```diff
@@ -20,55 +21,48 @@
-  <link rel="stylesheet" href="{{ $cdn_url_reveal }}/css/theme/{{ $theme }}.min.css">
+  <link rel="stylesheet" href="{{ $cdn_url_reveal }}/theme/{{ $theme }}.min.css" />
 
+  {{ printf "<script src=\"%s\" integrity=\"%s\" crossorigin=\"anonymous\"></script>" (printf $js.jQuery.url $js.jQuery.version) $js.jQuery.sri | safeHTML }}
```
jQuery appears to be redundant now that `fixMermaid` is vanilla JS?

dralmarzouq replied:
yes, I cannot remove since it will break slides without receiving the updated refactor.

don't forget this part as well:

```js
// document ready
  $(document).ready(function () {
    fixMermaid();
  });
```

dralmarzouq replied:
im assuming u will take care of this when merging the 2 PRs

gcushen commented on wowchemy/layouts/slides/baseof.html:
```diff
@@ -20,55 +21,48 @@
+  <link rel="stylesheet" href="{{ $cdn_url_reveal }}/reveal.min.css">
   {{- $theme := $.Param "slides.theme" | default "black" -}}
-  <link rel="stylesheet" href="{{ $cdn_url_reveal }}/css/theme/{{ $theme }}.min.css">
+  <link rel="stylesheet" href="{{ $cdn_url_reveal }}/theme/{{ $theme }}.min.css" />
```
We shouldn't close these tags with `/` in the HTML version used by Wowchemy.

gcushen commented on wowchemy/assets/js/wowchemy-slides.js:
```diff
@@ -0,0 +1,166 @@
+  }
+
+  // document ready
+  $(document).ready(function () {
```
Please use vanilla JS rather than jQuery

dralmarzouq replied:
alright .. but the fixmermkid is still there and needs extraction

gcushen commented on wowchemy/assets/js/wowchemy-slides.js:
```diff
@@ -0,0 +1,166 @@
+
+  {
+    {
+      /*  document.addEventListener("DOMContentLoaded", function(event) { 
```
Please remove commented code and refactor the PR so that it's ready for production use

dralmarzouq replied:
comments removed and beautified the file .. this part is vague though:

    refactor the PR so that it's ready for production use

gcushen commented on wowchemy/layouts/slides/baseof.html:
```diff
@@ -20,55 +21,48 @@
-      { src: '{{ $cdn_url_reveal }}/plugin/math/math.js', async: true },
-      // Enable exporting deck to PDF.
-      { src: '{{ $cdn_url_reveal }}/plugin/print-pdf/print-pdf.js', async: true },
-      // Enable speaker notes. Notes plugin must be loaded locally as CDN is missing `notes.html`.
```
I assume that speaker notes are now broken as you appear to have removed both the notes JS and HTML files but only added back in the notes JS via CDN?

dralmarzouq replied:
speaker notes work .. its now integrated into reveal.
This part enables it:

`<script src="{{ $cdn_url_reveal }}/plugin/notes/notes.js"></script>`

gcushen commented on wowchemy/test/config.yaml:
```diff
@@ -1,11 +0,0 @@
-title: Wowchemy Test
```
This file shouldn't be deleted...

dralmarzouq replied:
I replaced it with config.toml

some of the issues we faced were configuration related

they are now both included

gcushen commented on wowchemy/test/content/home/section.md:
```diff
@@ -1,9 +1,10 @@
 ---
```
Undo changes to this file (they are unnecessary)

gcushen commented on wowchemy/test/content/reveal.html:
```diff
@@ -0,0 +1,79 @@
+<!DOCTYPE html>
```
Please remove this file from the PR

dralmarzouq replied:
slides/index.md is still there

dralmarzouq commented:
@gcushen synced with upstream and now using utils files.

dralmarzouq commented:
@gcushen need to clean up the commit history .. please do not merge

dralmarzouq commented:
@gcushen fixed the PR and its commit history. Managed to duplicate all commits
now using fixMermaid from utils
I think we are set with this PR from my end

dralmarzouq commented:
PR should be now in sync with upstream

gcushen commented on wowchemy/assets/js/wowchemy-theming.js:
```diff
@@ -180,7 +180,7 @@ function renderThemeVariation(isDarkTheme, themeMode = 2, init = false) {
       console.debug('Initializing Mermaid with light theme');
       if (init) {
         /** @namespace window.mermaid **/
-        window.mermaid.initialize({theme: 'default', securityLevel: 'loose'});
```
Todo: add security level back

gcushen replied:
Please explain why you have changed the Mermaid security level from "loose" to "strict"?

gcushen replied:
@malmarz ?

gcushen commented on wowchemy/assets/js/wowchemy-slides.js:
```diff
@@ -0,0 +1,144 @@
+  //mermaid options
+  // mermaid: front matter configuration can be used to set mermaid options
+  // You can also use directives (see mermaid documentation)
+  var mermaidOptions =
```
refactor this to IF statement

gcushen commented:
# Todo

Remaining todo includes:

- [ ] Remove Mermaid security param from user options - https://github.com/wowchemy/wowchemy-hugo-modules/pull/2018/files#r546721818
- [ ] Add Mermaid security param back to Wowchemy Theming: https://github.com/wowchemy/wowchemy-hugo-modules/pull/2018/files#r550828149
- [ ] Refactor to use IF statement: https://github.com/wowchemy/wowchemy-hugo-modules/pull/2018/files#r550828354
- [ ] Remove jQuery: https://github.com/wowchemy/wowchemy-hugo-modules/pull/2018/files#r546876161
- [ ] Why are you proposing to send Reveal CDN URL as a JS param? Remove/refactor unless it's really needed: https://github.com/wowchemy/wowchemy-hugo-modules/pull/2018/files#r545006141
- [ ] Remove the duplicate config file you propose adding: https://github.com/wowchemy/wowchemy-hugo-modules/pull/2018/files#diff-3955202dbb75fab5f35e0092bdaf9dbd06f5561b74764b09d08287067b47d321
- [ ] Remove test/content/test/index.md - https://github.com/wowchemy/wowchemy-hugo-modules/pull/2018/files#diff-55d0f2cfdfcec540460113e76b317ee121205ba37269f4f774a6f11fe605856e
- [ ] Revert changes to test/go.mod and test/go.sum
- [ ] Revert changes to test/content/home/section.md

dralmarzouq commented on wowchemy/layouts/slides/baseof.html:
```diff
@@ -20,55 +21,44 @@
+    {{ printf "<script src=\"%s\" integrity=\"%s\" crossorigin=\"anonymous\" title=\"mermaid\"></script>" (printf $js.mermaid.url $js.mermaid.version) $js.mermaid.sri | safeHTML }}
+  {{ end }}
+
+  {{ $slidejs := resources.Get "js/wowchemy-slides.js" | js.Build (dict "params" (dict "slides" $.Params.slides "cdn_url_reveal" $cdn_url_reveal )) }}
```
this was used when we had theme selection in menu, removing as its not used anymore

dralmarzouq reviewed (commented):
> # Todo
> 
> Remaining todo includes:
> 
>     * [x]  Remove Mermaid security param from user options - https://github.com/wowchemy/wowchemy-hugo-modules/pull/2018/files#r546721818
> 
>     * [x]  Add Mermaid security param back to Wowchemy Theming: https://github.com/wowchemy/wowchemy-hugo-modules/pull/2018/files#r550828149
> 
>     * [x]  Refactor to use IF statement: https://github.com/wowchemy/wowchemy-hugo-modules/pull/2018/files#r550828354
> 
>     * [x]  Remove jQuery: https://github.com/wowchemy/wowchemy-hugo-modules/pull/2018/files#r546876161
> 
>     * [x]  Why are you proposing to send Reveal CDN URL as a JS param? Remove/refactor unless it's really needed: https://github.com/wowchemy/wowchemy-hugo-modules/pull/2018/files#r545006141
> 
>     * [x]  Remove the duplicate config file you propose adding: https://github.com/wowchemy/wowchemy-hugo-modules/pull/2018/files#diff-3955202dbb75fab5f35e0092bdaf9dbd06f5561b74764b09d08287067b47d321
> 
>     * [x]  Remove test/content/test/index.md - https://github.com/wowchemy/wowchemy-hugo-modules/pull/2018/files#diff-55d0f2cfdfcec540460113e76b317ee121205ba37269f4f774a6f11fe605856e

removed all new test content including slide examples.
Please note slides will not work correctly if using the current test config. I forgot which options needs to be included for it to work. But the config used in site templates should be fine.

>     * [x]  Revert changes to test/go.mod and test/go.sum

go.mod was fixed. didnt have history on go.sum, tried the test site and it works. 

>     * [x]  Revert changes to test/content/home/section.md

gcushen commented:
@dralmarzouq thanks! Please can you push your changes to the PR so that we can see them?

dralmarzouq commented:
> @dralmarzouq thanks! Please can you push your changes to the PR so that we can see them?

though I already did .. had an issue with authentication
should be pushed now

gcushen commented:
> > @dralmarzouq thanks! Please can you push your changes to the PR so that we can see them?
> 
> though I already did .. had an issue with authentication
> should be pushed now

`test/go.mod` and `test/go.sum` still haven't been reverted in the PR - they should not appear in the "Files changed" tab...

dralmarzouq commented:
> > > @dralmarzouq thanks! Please can you push your changes to the PR so that we can see them?
> > 
> > 
> > though I already did .. had an issue with authentication
> > should be pushed now
> 
> `test/go.mod` and `test/go.sum` still haven't been reverted in the PR - they should not appear in the "Files changed" tab...

ok I think its done correctly now ... files changed dropped from 11 to 9

I also synced with upstream .. without duplicating the commit history this time

malmarz commented:
please take note I'm fixing the mess that it my development pc settling on the malmarz username and profile

gcushen commented on wowchemy/assets/js/wowchemy-slides.js:
```diff
@@ -0,0 +1,146 @@
+// reveal configurations can be included in front matter under slides.reveal
+var pluginOptions = {}
+if(typeof params.slides.reveal_options === "undefined"){
+  pluginOptions = params.slides.reveal_options;
```
What is the purpose of checking for undefined and then setting the option to undefined? Did you mean to set it to `{}` from `pluginOptions`?

gcushen replied:
Did you mean **not** undefined?

malmarz replied:
yep .. that's what I meant 👍 
Not defined

gcushen replied:
The undefined checks are inconsistently implemented. Sometimes you have used `typeof`, sometimes you haven't. Please update all the undefined checks to be consistent.

gcushen commented on wowchemy/assets/js/wowchemy-slides.js:
```diff
@@ -0,0 +1,146 @@
+}
+
+
+pluginOptions = keysToCamel(pluginOptions);
```
You already set `pluginOptions` to `{}` above and now you're setting it again?

gcushen commented on wowchemy/assets/js/wowchemy-slides.js:
```diff
@@ -0,0 +1,146 @@
+
+  // make sure we have a menu configuration so we can set defaults
+  pluginOptions.menu = {}
+  if(typeof pluginOptions.menu === "undefined"){
```
Why are you checking if `pluginOptions.menu` is undefined when you just set it to `pluginOptions.menu = {}` in the line above?

malmarz replied:
should be removed because of the way we are now passing configuration to plugins .. there are also no defaults since chalkboard not included.

gcushen commented on wowchemy/assets/js/wowchemy-slides.js:
```diff
@@ -0,0 +1,146 @@
+  // You can also use directives (see mermaid documentation)
+  var mermaidOptions = {}
+  if(typeof params.slides.diagram_options === "undefined"){
+    mermaidOptions = params.slides.diagram_options;
```
```
  if(typeof params.slides.diagram_options === "undefined"){
    mermaidOptions = params.slides.diagram_options;
```

What is the purpose of checking for undefined and then also setting the pluginOptions option to undefined? Did you mean not undefined?

malmarz replied:
yet not defined .. was using a ternary operator and you requested to use if instead for readability I guess
copied the condition without modification

gcushen commented on wowchemy/assets/js/wowchemy-slides.js:
```diff
@@ -0,0 +1,146 @@
+      // mermaid slide index
+      shidx = Reveal.getIndices((slide = s))["h"];
+      if (
+        !diag.hasAttribute("data-processed") &&
```
Please comment each of these 4 checks so that future contributors understand the purpose of each check

gcushen commented on wowchemy/assets/js/wowchemy-slides.js:
```diff
@@ -0,0 +1,146 @@
+      background = s.slideBackgroundElement;
+      // render if we are 1 slide away horizontally
+      // current visible slide index
+      chidx = Reveal.getState()["indexh"];
```
what is the meaning of `chidx`? Is there a better name for it?

Same with `shidx`...

malmarz replied:
changed to more readable variable

gcushen commented on wowchemy/assets/js/wowchemy-slides.js:
```diff
@@ -0,0 +1,146 @@
+        background.style.display === "block" &&
+        shidx - chidx <= 1
+      )
+        return s;
```
what is the meaning of `s`?

malmarz replied:
renamed variable

gcushen commented on wowchemy/assets/js/wowchemy-slides.js:
```diff
@@ -0,0 +1,146 @@
+  }
+
+  function renderMermaidSlides() {
+    dSlides = Reveal.getSlides().filter(mermaidSlidesReadyToRender);
```
what is the meaning of `dSlides`?

malmarz replied:
renamed variable

malmarz commented:
improvement done and synced with upstream

gcushen commented on wowchemy/test/go.mod:
```diff
@@ -2,6 +2,6 @@ module github.com/wowchemy/wowchemy-hugo-modules/test
 go 1.15
 
-require github.com/wowchemy/wowchemy-hugo-modules/wowchemy master
+require github.com/wowchemy/wowchemy-hugo-modules/wowchemy v0.0.0-20210105225557-fe24b613d566
```
@malmarz you have pushed changes to `test/go.mod` again - please revert this. You can change the file locally if you want without committing changes to it and pushing it to the PR.

malmarz replied:
fixed

gcushen commented on wowchemy/layouts/slides/baseof.html:
```diff
@@ -23,55 +24,47 @@
-    let revealOptions = Object.assign({}, revealDefaults, revealPlugins);
-    Reveal.initialize(revealOptions);
-  </script>
+  <script src="{{ $cdn_url_reveal }}/reveal.min.js"></script>
```
@malmarz SRI (integrity and crossorigin) is missing from all these dependencies you added. Please fix.

malmarz replied:
done

gcushen commented on wowchemy/assets/js/wowchemy-slides.js:
```diff
@@ -0,0 +1,150 @@
+  if (isObject(o)) {
+    const n = {};
+
+    Object.keys(o).forEach((k) => {
```
You are defining functions using the arrow notation `(...) =>` in some places and `function (...)` in other places. Please be consistent and use the newer arrow notation when defining functions.

malmarz replied:
found middle ground for this .. since I see that you are using function in other files .. I followed the following rule of thumb:
- Use function when defining reusable functions
- Use arrow function for anonymous functions used in forEach and other methods that take a function as an argument

malmarz commented:
@gcushen any further updates to pr? It is it ready for merging?

gcushen commented:
> @gcushen any further updates to pr? It is it ready for merging?

Appears to be no response yet on a couple of review comments:

- https://github.com/wowchemy/wowchemy-hugo-modules/pull/2018/files#r550933338
- https://github.com/wowchemy/wowchemy-hugo-modules/pull/2018#discussion_r554145985

Also, it's best to install Yarn and Node and run the following JS lint and format commands (https://github.com/wowchemy/wowchemy-hugo-modules/blob/master/.github/contributing.md#where-to-start):

```
cd wowchemy
yarn install
yarn run lint
yarn run format
```

malmarz reviewed (commented):
@gcushen took care of linting and formatting issues .. as a result the following was changed:

- Added global: comment to identify the global variables in wowchemy-slides.js .... not sure if there is a better way to fix this
- changed all arrow functions to regular functions
- renamed some of the variables and functions and caught some naming related bugs
- did some format fixing to wowchemy-themeing.js

let me know if anything else is needed ... synced pr with upstream

gcushen commented:
Thanks for contributing to upgrade the slides capability @malmarz :)

gcushen commented:
Closes #1802

koutbo6 commented:
Was a pleasure

Keep up the good work

Next pr will likely be smoother .. I hope :)

> 
> On 20 Jan 2021, at 2:24 AM, Geo <notifications@github.com> wrote:
> 
> ﻿
> ----==_mimepart_60076a322d2f6_d591a0493882
> Content-Type: text/plain;
> charset=UTF-8
> Content-Transfer-Encoding: 7bit
> 
> Thanks for contributing to upgrade the slides capability @malmarz :)
> 
> -- 
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub:
> https://github.com/wowchemy/wowchemy-hugo-modules/pull/2018#issuecomment-763207556
> ----==_mimepart_60076a322d2f6_d591a0493882
> Content-Type: text/html;
> charset=UTF-8
> Content-Transfer-Encoding: 7bit
> 
> <p></p>
> <p>Thanks for contributing to upgrade the slides capability <a class="user-mention" data-hovercard-type="user" data-hovercard-url="/users/malmarz/hovercard" data-octo-click="hovercard-link-click" data-octo-dimensions="link_type:self" href="https://github.com/malmarz">@malmarz</a> :)</p>
> 
> <p style="font-size:small;-webkit-text-size-adjust:none;color:#666;">&mdash;<br />You are receiving this because you were mentioned.<br />Reply to this email directly, <a href="https://github.com/wowchemy/wowchemy-hugo-modules/pull/2018#issuecomment-763207556">view it on GitHub</a>, or <a href="https://github.com/notifications/unsubscribe-auth/AABULNX74STBDDMWRPAMGUDS2YIDFANCNFSM4UPTJQRA">unsubscribe</a>.<img src="https://github.com/notifications/beacon/AABULNRAWNWNWP2JU4DIYMLS2YIDFA5CNFSM4UPTJQRKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOFV6Z7BA.gif" height="1" width="1" alt="" /></p>
> <script type="application/ld+json">[
> {
> "@context": "http://schema.org",
> "@type": "EmailMessage",
> "potentialAction": {
> "@type": "ViewAction",
> "target": "https://github.com/wowchemy/wowchemy-hugo-modules/pull/2018#issuecomment-763207556",
> "url": "https://github.com/wowchemy/wowchemy-hugo-modules/pull/2018#issuecomment-763207556",
> "name": "View Pull Request"
> },
> "description": "View this Pull Request on GitHub",
> "publisher": {
> "@type": "Organization",
> "name": "GitHub",
> "url": "https://github.com"
> }
> }
> ]</script>
> ----==_mimepart_60076a322d2f6_d591a0493882--
